shared array: array.last() produces C error

<!-- You can use `v doctor` to fill up the next fields -->
OS: linux, Ubuntu 20.04.4 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/woody
vmodules: /home/woody/.vmodules
vroot: /home/woody/sources/v/v
vexe: /home/woody/sources/v/v/v
vexe mtime: 2022-08-06 08:41:02
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 372857c.4588bb4

Git version: git version 2.25.1
Git vroot status: weekly.2022.31-20-g4588bb44
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->

```
module main

fn main() {
	shared a := []int{}
	lock  {
		a << 1
		a << 2
	}
	rlock a {
		println(a.last())
	}
}

```

**What did you expect to see?**
No C error
 
**What did you see instead?**
```
woody@lappi:~/sources/v/bugs$ v shared_last.v
==================
/tmp/v_1000/shared_last.12850093158604570159.tmp.c:17031: error: cannot convert 'struct array *' to 'struct array'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1009,7 +1009,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			if !node.left.is_lvalue() {
 				g.write('ADDR($rec_cc_type, ')
 				has_cast = true
-			} else {
+			} else if node.name !in ['first', 'last', 'repeat'] {
 				g.write('&')
 			}
 		}

--- a/vlib/v/tests/shared_array_last_test.v
+++ b/vlib/v/tests/shared_array_last_test.v
@@ -1,0 +1,13 @@
+module main
+
+fn test_shared_array_last() {
+	shared a := []int{}
+	lock  {
+		a << 1
+		a << 2
+	}
+	rlock a {
+		println(a.last())
+		assert a.last() == 2
+	}
+}

